compiler/src/compiler.rs

```diff
@@ -524,4 +532,24 @@ LLVM module timings:
 
         Ok(exe)
     }
+
+    fn prepare(&mut self) -> Result<(), CompileError> {
+        let cur_ver = Version::parse(env!("CARGO_PKG_VERSION")).unwrap();
+        let iter = Manifest::all(&self.state.config)
+            .map_err(CompileError::Internal)?;
+
+        for manifest in iter {
+            let Some(ver) = manifest.minimum_inko_version() else { continue };
+
+            if ver > cur_ver {
+                return Err(CompileError::Internal(format!(
+                    "This project requires Inko {} or newer, \
+                    but the current version is {}",
+                    ver, cur_ver
+                )));
+            }
+        }
```

Comment:
I opted not to print the paths to the manifest here. Since the paths are hashed, including them isn't very useful, and we don't have the facilities to translate them back to package URIs. Thus, for the time being this error message will have to do.